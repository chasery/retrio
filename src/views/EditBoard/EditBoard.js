import React, { useContext, useState } from 'react';
import { useHistory, useParams } from 'react-router-dom';
import RetrioContext from '../../context/retrio-context';
import Header from '../../components/Header/Header';
import Form from '../../components/Form/Form';
import FormField from '../../components/FormField/FormField';
// import Error from '../../components/Error/Error';
import './EditBoard.css';

function EditBoard(props) {
  const context = useContext(RetrioContext);
  const history = useHistory();
  const { boardId } = useParams();

  const boardToEdit = context.boards.find((board) => board.id === boardId);

  const [team, setTeam] = useState(boardToEdit ? boardToEdit.team_id : '');
  const [boardName, setBoardName] = useState(
    boardToEdit ? boardToEdit.name : ''
  );
  // const [error, setError] = useState(null);

  const handleEditBoard = (e) => {
    e.preventDefault();

    context.editBoard(boardId, {
      name: boardName,
      team_id: team,
    });
    history.push(`/boards`);
  };

  const renderTeamOptions = (teams) => {
    return teams.map((team) => (
      <option key={team.id} value={team.id}>
        {team.name}
      </option>
    ));
  };

  return (
    <>
      <Header />
      <main role='main'>
        <section className='EditBoard'>
          <div className='EditBoard__wrapper'>
            <Form id='EditBoard' onSubmit={(e) => handleEditBoard(e)}>
              <div className='Form__header'>
                <h2>Edit Board</h2>
              </div>
              <div className='Form__body'>
                <p>Make changes to the Retrio board.</p>
                <div className='FormField'>
                  <label className='FormField__label' htmlFor={team}>
                    Team<span className='FormField__required'>*</span>
                  </label>
                  <div className='FormField__select'>
                    <select
                      id='team'
                      required
                      value={team}
                      onChange={(e) => setTeam(e.target.value)}
                    >
                      <option value=''>Select Team</option>
                      {renderTeamOptions(context.teams)}
                    </select>
                  </div>
                </div>
                <FormField
                  id='boardName'
                  label='Board Name'
                  type='text'
                  isRequired={true}
                  onChange={(e) => setBoardName(e.target.value)}
                  value={boardName}
                />
                {/* {error ? <Error message={error} /> : null} */}
                <div className='Form__controls'>
                  <button
                    className='Form__button secondary'
                    type='button'
                    onClick={() => history.push(`/boards`)}
                  >
                    Cancel
                  </button>
                  <button className='Form__button primary' type='submit'>
                    Edit Board
                  </button>
                </div>
              </div>
            </Form>
          </div>
        </section>
      </main>
    </>
  );
}

export default EditBoard;
