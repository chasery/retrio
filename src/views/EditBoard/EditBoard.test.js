import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EditBoard from './EditBoard';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    boardId: '800f3047-0054-4146-affsb7a-e06bf86f0868',
  }),
}));

describe('EditBoard component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<EditBoard />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
