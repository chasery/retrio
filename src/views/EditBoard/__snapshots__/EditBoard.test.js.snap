// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditBoard component renders without error 1`] = `
<Fragment>
  <Header />
  <main
    role="main"
  >
    <section
      className="EditBoard"
    >
      <div
        className="EditBoard__wrapper"
      >
        <Form
          id="EditBoard"
          onSubmit={[Function]}
        >
          <div
            className="Form__header"
          >
            <h2>
              Edit Board
            </h2>
          </div>
          <div
            className="Form__body"
          >
            <p>
              Make changes to the Retrio board.
            </p>
            <div
              className="FormField"
            >
              <label
                className="FormField__label"
                htmlFor=""
              >
                Team
                <span
                  className="FormField__required"
                >
                  *
                </span>
              </label>
              <div
                className="FormField__select"
              >
                <select
                  id="team"
                  onChange={[Function]}
                  required={true}
                  value=""
                >
                  <option
                    value=""
                  >
                    Select Team
                  </option>
                </select>
              </div>
            </div>
            <FormField
              id="boardName"
              isRequired={true}
              label="Board Name"
              onChange={[Function]}
              type="text"
              value=""
            />
            <div
              className="Form__controls"
            >
              <button
                className="Form__button secondary"
                onClick={[Function]}
                type="button"
              >
                Cancel
              </button>
              <button
                className="Form__button primary"
                type="submit"
              >
                Edit Board
              </button>
            </div>
          </div>
        </Form>
      </div>
    </section>
  </main>
</Fragment>
`;
